### ISSUE TITLE
cachedImageNamed method let APP crash in UIImage+ASConvenience.h

### ISSUE THREAD

**Issue by xiaomaogong** [facebook/AsyncDisplayKit#3228]
Any idea why this as_imageNamed cause app crash in 2.2 version of AsyncDisplayKit?

[__NSCFString subtype]: unrecognized selector sent to instance    ###0x197d1c00

Crashed: com.twitter.crashlytics.ios.exception
0  MyAPP                   0xf39971 CLSProcessRecordAllThreads + 119336
1  MyAPP                   0xf39971 CLSProcessRecordAllThreads + 119336
2  MyAPP                   0xf39869 CLSProcessRecordAllThreads + 119072
3  MyAPP                   0xf2d7af CLSHandler + 69734
4  MyAPP                   0xf3825b __CLSExceptionRecord_block_invoke + 113426
5  libdispatch.dylib              0x22f3d80f _dispatch_client_callout + 22
6  libdispatch.dylib              0x22f4901d _dispatch_barrier_sync_f_invoke + 56
7  MyAPP                   0xf37c5d CLSExceptionRecord + 111892
8  MyAPP                   0xf37a85 CLSExceptionRecordNSException + 111420
9  MyAPP                   0xf3759b CLSTerminateHandler() + 110162
10 libc++abi.dylib                0x22b5ce17 std::__terminate(void (*)()) + 78
11 libc++abi.dylib                0x22b5c5f5 __cxxabiv1::exception_cleanup_func(_Unwind_Reason_Code, _Unwind_Exception*) + 110
12 libobjc.A.dylib                0x22b6aeeb objc_exception_throw + 250
13 CoreFoundation                 0x233d52b5 __methodDescriptionForSelector + 186
14 CoreFoundation                 0x233d2ee1 ___forwarding___ + 700
15 CoreFoundation                 0x232fe238 _CF_forwarding_prep_0 + 24
16 CoreUI                         0x277ade19 -[CUINamedImage _cacheRenditionProperties] + 232
17 CoreUI                         0x27797b7b -[CUINamedLookup initWithName:usingRenditionKey:fromTheme:] + 146
18 CoreUI                         0x277b9b9f -[CUICatalog imageWithName:scaleFactor:deviceIdiom:deviceSubtype:sizeClassHorizontal:sizeClassVertical:] + 170
19 UIKit                          0x2827f881 -[UIImageAsset imageWithTraitCollection:] + 516
20 UIKit                          0x2828074f -[_UIPathLazyImageAsset imageWithTraitCollection:] + 346
21 UIKit                          0x28179d45 -[_UIAssetManager imageNamed:withTrait:] + 636
22 UIKit                          0x27c283dd +[UIImage imageNamed:inBundle:compatibleWithTraitCollection:] + 172
23 MyAPP                   0x9440f9 cachedImageNamed (UIImage+ASConvenience.m:43)
24 MyAPP                   0x451f1b -[MyCellNode initUI] (MyCellNode.m:51)
25 MyAPP                   0x451d67 -[MyCellNode init] (MyCellNode.m:40)
26 MyAPP                   0x789e29 -[MyCellNode initUI] (MyCellNode.m:77)
27 MyAPP                   0x789d41 -[MyCellNode initNodeWithViewModel:] (MyCellNode.m:69)
28 MyAPP                   0x1485c7 -[MyCollectionlNode getCollectionNode:indexPath:] (MyCollectionlNode.m:195)
29 MyAPP                   0x6b7e17 __70-[MyCollectionlNode collectionView:nodeBlockForItemAtIndexPath:]_block_invoke (MyCollectionlNode.m:181)
30 MyAPP                   0x8a80a3 __56-[ASCollectionView dataController:nodeBlockAtIndexPath:]_block_invoke_3.553 (ASCollectionView.mm:1495)
31 MyAPP                   0x8e4091 -[ASIndexedNodeContext node] (ASIndexedNodeContext.mm:51)
32 MyAPP                   0x8afb79 __45-[ASDataController _layoutNodesFromContexts:]_block_invoke (ASDataController.mm:212)
33 MyAPP                   0x8b6099 ___ZL15ASDispatchApplymPU28objcproto17OS_dispatch_queue8NSObjectjU13block_pointerFvmE_block_invoke (ASDispatch.h:26)
34 libdispatch.dylib              0x22f3d823 _dispatch_call_block_and_release + 10
35 libdispatch.dylib              0x22f4c5e9 _dispatch_root_queue_drain + 1560
36 libdispatch.dylib              0x22f4bfcd _dispatch_worker_thread3 + 96
37 libsystem_pthread.dylib        0x23101b29 _pthread_wqthread + 1024
38 libsystem_pthread.dylib        0x23101718 start_wqthread + 8

**Comment by nguyenhuy** [comment]
@xiaomaogong Can you reproduce the crash without `UIImage+ASConvenience`? Please make sure your asset is correct for the specific iOS version on which the crash occurred. Some people got crashes in `CUINamedImage` because their assets were incompatible. 

Feel free to re-open this issue if you are sure the problem is in `UIImage+ASConvenience` and ideally with a reproducible sample project. Thank you!

**Comment by xiaomaogong** [comment]
@nguyenhuy 
We can't reproduce this issue , we got this crash log from fabric.
But we just found all images cause this crash are stored directly in files not assets file.
Meanwhile we didn't get these crashes before use [UIImage imageNamed:] , which means i suspect the crash reason might relative to the call to [UIImage imageNamed:inBundle:compatibleWithTraitCollection:] in cachedImageNamed (UIImage+ASConvenience.m) when these images are not stored in assets file.
Still thanks for your explanation for this issue.
